source/code/include/playfab/PlayFabError.h.ejs

```diff
@@ -42,4 +42,29 @@ namespace PlayFab
 
     typedef std::function<void(const PlayFabError& error, void* customData)> ErrorCallback;
     typedef std::function<void(std::exception exception)> ExceptionCallback;
+
+    enum class PlayFabExceptionCode
+    {
+        AuthContextRequired,
+        DeveloperKeyNotSet,
+        EntityTokenNotSet,
+        NotLoggedIn,
+        PluginAmbiguity,
+        PluginNotFound,
+        ThreadMisuse,
+        TitleNotSet,
+    };
+
+    class PlayFabException : public std::runtime_error
```

Comment:
>runtime_error [](start = 41, length = 13)

why are we not deriving from std::exception ? will catch(...) still handle this ?

Reply:
std::runtime_error inherits from std::exception.
[http://www.cplusplus.com/reference/stdexcept/runtime_error/](http://www.cplusplus.com/reference/stdexcept/runtime_error/)
runtime_error is more appropriate for our exceptions, and in fact, the existing ones were already std::runtime_error


---
In reply to: [343288358](https://github.com/PlayFab/XPlatCoreTemplate/pull/39#discussion_r343288358) [](ancestors = 343288358)